src/main/java/com/kylas/sales/workflow/domain/workflow/action/reassign/ReassignDetail.java

```
@@ -17,10 +18,18 @@
 
   private final Long entityId;
   private final Long ownerId;
+  @JsonIgnore
+  private final EntityType entityType;
 
   @Override
   @JsonIgnore
   public String getEventName() {
+    switch (this.entityType) {
+      case LEAD:
+        return "workflow.lead.reassign";
+      case DEAL:
+        return "workflow.deal.reassign";
+    }
     return "workflow.lead.reassign";
```

Instead of returning default value, just throw custom exception to avoid misleading of event to be trigger